### DISCUSSION TITLE
Completely disable sky

### DISCUSSION THREAD

**Discussion by UberMouse** [maplibre/maplibre-gl-js#4512] [General]
Hi,

Just wondering if it's possible to completely disable the new sky/fog/horizon styling? I've attempted to do this by setting the the sky/fog/horizon colors to a completely see-through color (`rgba(0, 0, 0, 0)`) but elements of it are still visible.

**Comment by HarelM** [comment]
Probably related to this comment: 
https://github.com/maplibre/maplibre-gl-js/pull/3645#discussion_r1701067263

It might be a missing implementation...

**Reply by UberMouse** [reply]
Hmm the video in that post implies that `setSky(undefined)` is what we want, but it doesn't seem to work when combined with `setStyle`. I'll see if I can make a reproduction and raise an issue for it tomorrow. We drive everything off generating a custom map style and calling `setStyle` so we need to be able to turn off sky that way.

**Reply by HarelM** [reply]
When using `setSky` there's a diff operation that happens and it might be that this diff is not handling the change correctly.
In any case, an issue with a reproduction would be the best way forward.
Thanks for looking into this!

**Reply by UberMouse** [reply]
Turns out it's not related to `setStyle`, I just figured that was the difference from the example video. The actual difference is having terrain enabled, can't disable horizon/fog rendering when terrain is enabled

https://github.com/maplibre/maplibre-gl-js/issues/4517
